keras_hub/src/models/yolo_v8/yolo_v8_label_encoder.py

```diff
@@ -0,0 +1,257 @@
+        alignment_metrics *= anchors_matched_gt_box
+        max_alignment_per_gt_box = ops.max(
+            alignment_metrics, axis=-1, keepdims=True
+        )
+        max_overlap_per_gt_box = ops.max(overlaps, axis=-1, keepdims=True)
+
+        normalized_alignment_metrics = ops.max(
+            alignment_metrics
+            * max_overlap_per_gt_box
+            / (max_alignment_per_gt_box + self.epsilon),
+            axis=-2,
+        )
+        class_labels *= normalized_alignment_metrics[:, :, None]
+
+        # On TF backend, the final "4" becomes a dynamic shape so we include
+        # this to force it to a static shape of 4. This does not actually
+        # reshape the Tensor.
+        bbox_labels = ops.reshape(bbox_labels, (-1, num_anchors, 4))
+        return (
+            ops.stop_gradient(bbox_labels),
+            ops.stop_gradient(class_labels),
+            ops.stop_gradient(
+                # ops.cast(gt_box_matches_per_anchor > -1, "float32")
+                ops.cast(gt_box_matches_per_anchor > 0, "float32")
+            ),
+        )
+
+    def call(
+        self, scores, decode_bboxes, anchors, gt_labels, gt_bboxes, gt_mask
+    ):
+        """Computes target boxes and classes for anchors.
+
+        Args:
+            scores: a Float Tensor of shape (batch_size, num_anchors,
+                num_classes) representing predicted class scores for each
+                anchor.
+            decode_bboxes: a Float Tensor of shape (batch_size, num_anchors, 4)
+                representing predicted boxes for each anchor.
+            anchors: a Float Tensor of shape (batch_size, num_anchors, 2)
+                representing the xy coordinates of the center of each anchor.
+            gt_labels: a Float Tensor of shape (batch_size, num_gt_boxes)
+                representing the classes of ground truth boxes.
+            gt_bboxes: a Float Tensor of shape (batch_size, num_gt_boxes, 4)
+                representing the ground truth bounding boxes in xyxy format.
+            gt_mask: A Boolean Tensor of shape (batch_size, num_gt_boxes)
+                representing whether a box in `gt_bboxes` is a real box or a
+                non-box that exists due to padding.
+
+        Returns:
+            A tuple of the following:
+                - A Float Tensor of shape (batch_size, num_anchors, 4)
+                    representing box targets for the model.
+                - A Float Tensor of shape (batch_size, num_anchors, num_classes)
+                    representing class targets for the model.
+                - A Boolean Tensor of shape (batch_size, num_anchors)
+                    representing whether each anchor was a match with a ground
+                    truth box. Anchors that didn't match with a ground truth
+                    box should be excluded from both class and box losses.
+        """
+        if isinstance(gt_bboxes, tf.RaggedTensor):
```

Comment:
To avoid importing TF you can use 

```
def is_tensorflow_ragged(value):
    if hasattr(value, "__class__"):
        return (
            value.__class__.__name__ == "RaggedTensor"
            and "tensorflow.python." in str(value.__class__.__module__)
        )
    return False
```